File: game-server/src/main/java/org/forfun/mmorpg/game/database/config/domain/ConfigFunction.java
package org.forfun.mmorpg.game.database.config.domain;

import jforgame.data.annotation.DataTable;
import jforgame.data.annotation.Id;
import jforgame.data.annotation.Index;
import lombok.Getter;
import org.apache.commons.lang3.math.NumberUtils;
import org.forfun.mmorpg.game.function.model.FunctionOpenType;

@DataTable(name = "configfunction")
@Getter
public class ConfigFunction {
	
	@Id
	private int id;
	
	/**
	 * 开启类型 {@link FunctionOpenType#getType()}
	 */
	private FunctionOpenType openType;
	
	/**
	 * 开启类型对应的参数
	 */
	private String openParams;

	public int getOpenMainParam() {
		return NumberUtils.toInt(openParams);
	}

}

File: game-server/src/main/java/org/forfun/mmorpg/game/database/config/domain/ConfigFunction.kt
package org.forfun.mmorpg.game.database.config.domain

import jforgame.data.annotation.DataTable
import jforgame.data.annotation.Id
import org.apache.commons.lang3.math.NumberUtils
import org.forfun.mmorpg.game.function.model.FunctionOpenType

@DataTable(name = "configfunction")
class ConfigFunction {

    @Id
    var id: Int = 0

    /**
     * 开启类型 {@link FunctionOpenType#getType()}
     */
    var openType: FunctionOpenType? = null

    /**
     * 开启类型对应的参数
     */
    var openParams: String? = null

    fun getOpenMainParam(): Int {
        return NumberUtils.toInt(openParams)
    }
}
